W-19473121-hsa-component-partial-kt (#322)
mulesoft/docs-hosting: hybrid-standalone/modules/ROOT/pages/monitoring-dashboards-hybrid.adoc

= Monitoring Dashboards for Servers, Server Groups, and Clusters

Monitor your hybrid standalone infrastructure through comprehensive performance dashboards that provide real-time visibility into server health, resource utilization, and system performance across your distributed runtime environment.

== Overview

> Runtime Manager provides comprehensive infrastructure monitoring dashboards for your hybrid standalone environment, enabling you to:

Reviewer: ```suggestion
Runtime Manager provides comprehensive infrastructure monitoring dashboards for your hybrid standalone environment, so you can:
```

* View aggregated metrics across server groups and clusters
* Identify performance bottlenecks and system health issues
* Make data-driven decisions for infrastructure scaling and maintenance

[NOTE]
====
To view infrastructure dashboards, you must have the Runtime Manager Read Servers permission for the target servers.

Anypoint Platform Private Cloud Edition doesn't support the Anypoint Monitoring dashboard feature.
====

Runtime Manager provides these infrastructure monitoring dashboards:

* <<Server Monitoring Dashboard,Server Monitoring Dashboard>> +
Individual server performance metrics including CPU, memory, JVM heap usage, garbage collection, and detailed memory breakdown
* <<Server Group or Cluster Monitoring Dashboard,Server Group and Cluster Monitoring Dashboard>> +
Aggregated performance metrics for servers organized in groups or clusters with comparative analysis capabilities

== Server Monitoring Dashboard

Monitor individual server performance through comprehensive dashboards that provide real-time visibility into system resources, JVM metrics, and operational health indicators.

> === Overview

Reviewer: I don't think it's a good idea to have the same title twice in the same page

> === Overview

Author: ```suggestion

```

==== JVM Memory Architecture

Understanding JVM memory organization is essential for effective performance monitoring:

* Heap Memory Pools
** <<eden,Eden Space>>
** <<survivor,Survivor Space>>
** <<old_gen,Old Generation>>

* Non-Heap Memory Areas
** <<code_cache,Code Cache>>
** <<compressed_class,Compressed Class Space>>
** <<metaspace,Metaspace>>

[[eden]]
Eden Space::
Monitor the primary object allocation area for young generation garbage collection optimization:
+
image::monitoring-dashboards-eden.png[Monitoring dashboard for Eden Space]
+
Eden space metrics help you:
* Optimize object allocation patterns
* Tune young generation garbage collection frequency
* Identify excessive short-term object creation

[[survivor]]
Survivor Space::
Track intermediate object storage for garbage collection lifecycle analysis:
+
image::monitoring-dashboards-survivor.png[Monitoring dashboard for Survivor Space]
+
Survivor space monitoring enables:
* Analysis of object promotion patterns to old generation
* Optimization of survivor space sizing for GC efficiency
* Detection of objects with intermediate lifespans

[[old_gen]]

> Old Generation::

Reviewer: ```suggestion
Old Generation:
```

Reviewer: (globally) Are "::" on purpose?

=== Access Group and Cluster Dashboards

To monitor aggregated performance metrics for your server infrastructure:

. In Runtime Manager, click *Servers*.
. Click the server group or cluster name to display the aggregated dashboard.
. Select specific servers from the list to display comparative metrics in the details pane.

=== Dashboard Navigation

* Time Range Selection

Monitor performance trends across different time periods:

image::monitoring-dashboards-cbd92.png[Monitoring dashboard for a server group or cluster]

Choose from available time ranges:
* **3 Hours**: Real-time infrastructure monitoring
* **24 Hours**: Recent performance analysis  
* **7 Days**: Weekly infrastructure trends
* **2 Weeks**: Historical capacity analysis

* Server-Focused View

Access server-specific performance data within the group context:

image::monitoring-dashboards-6fe9f.png[Monitoring dashboard for a server group or cluster]

From the group dashboard, navigate to individual server dashboards by clicking server names to access detailed performance metrics and troubleshooting information.

=== Troubleshooting Server Group Monitoring

> If monitoring metrics do not appear immediately after configuring Anypoint Monitoring for servers in a server group, follow this systematic installation process:

Reviewer: ```suggestion
If monitoring metrics don't appear immediately after configuring Anypoint Monitoring for servers in a server group, follow this systematic installation process:
```

. In Anypoint Platform go to *Runtime Manager* > *Servers* > *Add Server* to create an Anypoint Runtime Manager token.
. Execute the command provided in the *Add Server* screen.
. In Anypoint Platform go to *Runtime Manager* > *Servers* > *Create Group* to create the server group.
. Start Mule for all the servers to be added to the server group by using the command `./mule start` from the `$MULE_HOME/bin` directory.
. In Anypoint Platform go to *Runtime Manager* > *Servers*, click the server group name, and *Add Server* to add all the servers to the group.
. Deploy an API to the server group.
. Stop any running filebeat process in all the servers of the server group and run the `./install` command from the `$MULE_HOME/am/bin` directory in all the servers of the server group.
. In Anypoint Platform go to *Runtime Manager* > *Servers*, click the server group name, and restart all the servers in the server group via *Actions* > *Restart*.
. Run `./setup` from the `$MULE_HOME/am/bin` for all the servers in the server group.
+
If after this step the metrics don't show up, repeat the last two steps.

== See Also

> * xref:monitoring-hybrid-standalone.adoc[Monitoring Hybrid Standalone Instances]

Reviewer: Is it necessary to use hard-coded titles?
